### PR TITLE
feature: show notice to donor when payment method does not support subscription

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

I added a notice which displays when a donor selected a payment method that does not support recurring donations.

## Visuals

image

## Testing Instructions

You should see a notice on the donation form when creating a subscription with payment gateway registered with new gateway API but does not have a subscription module.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/LegacySubscriptions/Notices/PaymentGatewayGatewayNotSupportSubscription.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            if (
+                give_recurring_is_donation_recurring($legacyFormData) &&
+                !$registeredGateway->supportsSubscriptions()
+            ) {
+                give_set_error(
+                    'payment-gateway-not-support-subscription',
+                    sprintf(
+                    /* translators: 1. Payment gateway name.*/
+                        esc_html__(
+                            'Currently, we are not supporting recurring donations with %1$s.',
```
Comment (Benunc) [inline src/LegacySubscriptions/Notices/PaymentGatewayGatewayNotSupportSubscription.php:33]:
```suggestion
                            'Recurring Donations are not supported with %1$s.',
```

File: src/LegacySubscriptions/Notices/PaymentGatewayGatewayNotSupportSubscription.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                            'give'
+                        ),
+                        $registeredGateway->getName()
```
Comment (Benunc) [inline src/LegacySubscriptions/Notices/PaymentGatewayGatewayNotSupportSubscription.php:36]: make sure this is the Display name, not the internal name. If the user has given the gateway a custom label, that should display here.

**Review by Benunc — CHANGES_REQUESTED** [review]

File: src/LegacySubscriptions/Notices/PaymentGatewayGatewayNotSupportSubscription.php (hunk 2 above)
Comment (ravinderk) [reply src/LegacySubscriptions/Notices/PaymentGatewayGatewayNotSupportSubscription.php]: fixed: https://github.com/impress-org/givewp/compare/8be0f41a052b...bd06545ed633

**Review by jonwaldstein — CHANGES_REQUESTED** [review]
@ravinderk we need to think through this more.  This would be very detrimental to recurring customers because it would prevent any existing gateway's recurring donation functionality _except_ the ones defined in our gateway api that happen to support subscriptions (which currently is 0).  

Remember, currently gateways are able to exist in core using the gateway api and recurring as legacy and are working properly - so we can't just prevent that from happening.

I see where you were going with this but it's just too soon to force the gateway api on our recurring gateways.  There will be a time in the near future when all gateways are using gateway api which at that point we can determine what the best way is to go about this.

**Comment by ravinderk** [comment]
> @ravinderk we need to think through this more. This would be very detrimental to recurring customers because it would prevent any existing gateway's recurring donation functionality _except_ the ones defined in our gateway api that happen to support subscriptions (which currently is 0).
> 
> Remember, currently gateways are able to exist in core using the gateway api and recurring as legacy and are working properly - so we can't just prevent that from happening.
> 
> I see where you were going with this but it's just too soon to force the gateway api on our recurring gateways. There will be a time in the near future when all gateways are using gateway api which at that point we can determine what the best way is to go about this.

@jonwaldstein I added a fix for this. Please review this: https://github.com/impress-org/givewp/compare/bd06545ed633...c821893aea90

**Comment by ravinderk** [comment]
> @ravinderk actually this PR is not necessary because we already have the functionality in place. If you look at `AddPaymentGatewayModulesToLegacyList` we are adding all gateways that support subscriptions to the `give_recurring_available_gateways` filter. That filter is being used `give_recurring_checkout_errors` to determine if a gateway is supported.
> 
> Screen Shot 2022-06-09 at 9 21 24 AM
> 
> Screen Shot 2022-06-09 at 9 16 41 AM

@jonwaldstein I thought, it will be good to add this to GiveWP core, but we can do it later. it is not required at this moment.

**Pull request closed without merging**